### PR TITLE
python-updates - major updates - 2023-02-27

### PR DESCRIPTION
**Pull request by mweinelt**
Another round of major updates, since it has been two months since the last one.

Jobset:
- https://hydra.nixos.org/jobset/nixpkgs/python-updates#tabs-evaluations

Reports:
- https://malob.github.io/nix-review-tools-reports/nixpkgs:python-updates/nixpkgs_python-updates_1791426.html

Dropped updates:
- elasticsearch 8.x, breaks elasticsearch-dsl, which seems abandoned, but is still widely used
- sphinx >6 fails to import extensions, needs investigation
- bokeh, panel still isn't ready, also https://github.com/NixOS/nixpkgs/pull/215406
- catalogue, 2.1.0 was yanked
- tensorboard 2.12, went out of sync with tensorflow

Needs help:
- aws-sam-translator fails its tests with NoRegionError in botocore
  - disabled all tests for now.

TODO:
- [x] Collect open PRs
- [x] Bulk updates
- [x] Early fixes
- [x] Start Evaluation https://hydra.nixos.org/jobset/nixpkgs/python-updates

---

Closes #218443

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
> sphinx fails to import certain extension on their latest version @SuperSandro2000
> * sphinx-applehelp 1.0.4 (reverted)
> * sphinx-htmlhelp 2.0.1 (reverted)

That is already the case for certain other extensions. I don't know how to properly fix it but adding them to the sphinx build itself somehow made it work. Probably sphinx does not handle well our usecase.

**Comment by kirillrdy** [comment]
fix awsebcli https://github.com/NixOS/nixpkgs/pull/218872

**Comment by dotlambda** [comment]
Looks like we won't be able to make Numba work with NumPy 1.24: https://github.com/numba/numba/issues/8464
cc @FRidh

**Comment by FRidh** [comment]
Then we should wait again with the numpy update.

File: pkgs/development/python-modules/pydocstyle/default.nix [hunk 1]
```diff
@@ -1,32 +1,44 @@
...
+  ];
+
   propagatedBuildInputs = [
     snowballstemmer
+  ] ++ lib.optionals (pythonOlder "3.11") [
+    tomli
+  ];
+
+  passthru.optional-dependencies.toml = [
+    tomli
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pydocstyle/default.nix:36]: This is duplicated, we could just condense this to always include tomli

File: pkgs/development/python-modules/flask-babel/default.nix [hunk 2]
```diff
@@ -32,6 +32,10 @@ buildPythonPackage rec {
     hash = "sha256-bHsB1f7dbZW4k8JteyZOwVCgWRDZMu21XdMcjM5NYjk=";
   };
 
+  patches = [
+    ./fix-tests-with-babel-12-2.diff
```
Comment (dotlambda) [inline pkgs/development/python-modules/flask-babel/default.nix:36]: @SuperSandro2000 You forgot to commit the file.

File: pkgs/top-level/python-aliases.nix [hunk 3]
```diff
@@ -50,6 +50,7 @@ mapAliases ({
   bt_proximity = bt-proximity; # added 2021-07-02
   BTrees = btrees; # added 2023-02-19
   carrot = throw "carrot has been removed, as its development was discontinued in 2012"; # added 2022-01-18
+  cchardet = faust-cchardet; # added 2023-02-14
```
Comment (mweinelt) [inline pkgs/top-level/python-aliases.nix:53]: On second thought, won't that bite us with regards to packages relying on aliases?

```
❯ rg cchardet -l
pkgs/development/tools/poetry2nix/poetry2nix/overrides/build-systems.json
pkgs/development/python-modules/cchardet/default.nix
pkgs/development/python-modules/aiohttp/default.nix
pkgs/development/python-modules/surepy/default.nix
pkgs/development/python-modules/nextcord/default.nix
pkgs/development/python-modules/vulcan-api/default.nix
pkgs/development/python-modules/clevercsv/default.nix
pkgs/applications/misc/calibre/default.nix
pkgs/servers/home-assistant/appdaemon.nix
pkgs/tools/backup/grab-site/default.nix
pkgs/tools/security/xcat/default.nix
pkgs/tools/security/cve-bin-tool/default.nix
```

Comment (dotlambda) [reply pkgs/top-level/python-aliases.nix]: Yes, I forgot to change the references.

File: pkgs/development/python-modules/blis/default.nix [hunk 4]
```diff
@@ -12,14 +12,14 @@ buildPythonPackage rec {
   pname = "blis";
   # Do not update to BLIS 0.9.x until the following issue is resolved:
   # https://github.com/explosion/thinc/issues/771#issuecomment-1255825935
-  version = "0.7.9";
+  version = "0.9.1";
```
Comment (danieldk) [inline pkgs/development/python-modules/blis/default.nix:15]: BLIS 0.9.1 is broken, see: https://github.com/NixOS/nixpkgs/pull/207070

**Comment by laikq** [comment]
Please revert or amend commit a8d46a2: package `pypdf2` should not be updated to version 3, it breaks `pdfposter` and probably other things.

**Comment by mweinelt** [comment]
As `pdfposter` is not the only consumer of `pypdf2` this needs more investigation to understand if that is feasible. As it stands it cannot be reverted easily.

**Pull request merged**